File: iOS/PanoAudioChat/PanoAudioChatSDK/Source/Sections/AudioRoom/View/ApplyUserList/PanoApplyAlertView.m
//
//  PanoApplyAlertView.m
//  PanoAudioChatSDK
//
//  Created by pano on 2021/7/19.
//

#import "PanoApplyAlertView.h"
#import "UIColor+Extension.h"
#import "UIImage+Extension.h"

@implementation PanoApplyAlertView {
    UIView *contentView;
}

- (void)initViews {
    
    self.backgroundColor = [UIColor clearColor];
    countBtn = [UIButton buttonWithType:UIButtonTypeCustom];
    UIImage *image = [UIImage pano_imageWithColor:[UIColor pano_colorWithHexString:@"#EC2C2C"]];
    
    [countBtn setBackgroundImage:image forState: UIControlStateNormal];
    [countBtn setBackgroundImage:image forState: UIControlStateNormal];
    countBtn.layer.masksToBounds = true;
    countBtn.layer.cornerRadius = 12.5;
    
    countLabel = [UILabel new];
    countLabel.textColor = [UIColor whiteColor];
    countLabel.textAlignment = NSTextAlignmentCenter;
    [countBtn addSubview:countLabel];
    
    [self addSubview:countBtn];
}

- (void)layoutSubviews{
    [super layoutSubviews];
}
    
- (void)initConstraints {

    [countBtn mas_makeConstraints:^(MASConstraintMaker *make) {
        make.size.mas_equalTo(CGSizeMake(25, 25));
        make.centerX.mas_equalTo(0);
        make.bottom.mas_equalTo(0);
    }];
    
    [countLabel mas_makeConstraints:^(MASConstraintMaker *make) {
        make.edges.mas_equalTo(0);
    }];
}

@end

File: iOS/PanoAudioChat/PanoAudioChat/Classes/Sections/AudioRoom/View/ApplyUserList/PanoApplyAlertView.swift
//
//  PanoApplyAlertView.swift
//  PanoAudioChat
//
//  Created by wenhua.zhang on 2020/11/3.
//

import UIKit

class PanoApplyAlertView: PanoBaseView {
    
    var countBtn: UIButton!
    var contentView: UIView!
    var countLabel: UILabel!
    override func initViews() {
        
        self.backgroundColor = UIColor.clear
        
//        contentView = UIView(frame: CGRect(x: 0, y: 0, width: PanoAppWidth, height: 20))
//        contentView.backgroundColor = UIColor.white
//        self.addSubview(contentView)
        
        countBtn = UIButton()
        let image = UIImage(color: UIColor("#EC2C2C"))
        countBtn.setBackgroundImage(image, for: .normal)
        countBtn.setBackgroundImage(image, for: .selected)
        countBtn.layer.masksToBounds = true
        countBtn.layer.cornerRadius = 12.5;
        
        countLabel = UILabel()
        countLabel.textColor = UIColor.white
        countLabel.textAlignment = .center
        countBtn.addSubview(countLabel)
        
        self.addSubview(countBtn)
    }
    
    override func layoutSubviews() {
        super.layoutSubviews()
        self.roundCorners(corners: [.bottomLeft, .bottomRight], radius: 15)
    }
        
    override func initConstraints() {
//        contentView.snp.makeConstraints { (make) in
//            make.left.right.top.equalToSuperview()
//            make.height.equalTo(20);
//        }
//        
        countBtn.snp.makeConstraints { (make) in
            make.size.equalTo(CGSize(width: 25, height: 25))
            make.centerX.equalToSuperview()
            make.bottom.equalToSuperview()
        }
        
        countLabel.snp.makeConstraints { (make) in
            make.edges.equalToSuperview()
        }
    }
    
    func updateAlertView(pushed: Bool) {
//        if pushed {
//            contentView.snp.makeConstraints { (make) in
//                make.top.left.right.equalToSuperview()
//                make.height.equalTo(20);
//            }
//        } else {
//            contentView.snp.makeConstraints { (make) in
//                make.top.centerX.equalToSuperview()
//                make.height.equalTo(20);
//                make.width.equalTo(184)
//            }
//        }
    }
}
